test/tests.py

```diff
@@ -85,24 +90,33 @@ def dmlc_reaper_args(exitcode_file, timeout_multiplier = 1):
 
 common_cflags = ["-O2", "-std=gnu99", '-Wall', '-Werror', '-Wpointer-arith',
                  '-Wwrite-strings', '-Wformat-nonliteral',]
+cc = [os.environ.get('DMLC_CC')]
 if is_windows():
-    cc = [testparams.mingw_cc()]
+    if not cc[0]:
+        try:
+            cc = [testparams.mingw_cc()]
+            cc_not_found = not os.path.exists(cc[0])
+        except FileNotFoundError:
```

Comment:
insufficient. Looking at the implementation, seems that mingw_cc can throw a bare Exception. And that it cannot return None.